terror/terror.go

```diff
@@ -179,6 +183,14 @@ func (e *Error) Code() ErrCode {
 	return e.code
 }
 
+// SetWorkaround is a decorator like method which add a workaround to
+// error which is convenient for user to search.
+func (e *Error) SetWorkaround(workaround string) *Error {
+	e.workaround = workaround
+	errCodeMap[e.code] = e
```

Comment:
If some `Error` does not have been called `SetWorkaround` then it can not be exported by `ExportErrorCodeAndWorkaround`. Is it by design?

Reply:
Yes, but I find  this is a bad design, we can't find whether we lose some error workaround. How about add this code into `New` function?

Reply:
We could add errors into `errCodeMap` in the `new` function so there is no missing, `SetWorkaround` could be an option for refactoring work.